### PR TITLE
Enable loadable modules mtl

### PR DESCRIPTION
**Pull request by jxstelter**
This patch will enable lib_manager and iadk_modules support for MTL platform.

### CODE REVIEW

File: src/audio/module_adapter/module/iadk_modules.c [hunk 1]
```diff
@@ -245,12 +245,13 @@ static int iadk_modules_set_configuration(struct processing_module *mod, uint32_
...
-					  uint32_t data_offset_size, const uint8_t *fragment,
+					  uint32_t *data_offset_size, uint8_t *fragment,
 					  size_t fragment_size)
 {
+	enum module_cfg_fragment_position pos = MODULE_CFG_FRAGMENT_SINGLE;
+
 	return iadk_wrapper_get_configuration(mod->priv.module_adapter, config_id, pos,
-					      data_offset_size, fragment, fragment_size);
+					  *data_offset_size, fragment, fragment_size);
 }
```
Comment (lyakh) [inline src/audio/module_adapter/module/iadk_modules.c:255]: how about just
```
return iadk_wrapper_get_configuration(mod->priv.module_adapter, config_id,
					  MODULE_CFG_FRAGMENT_SINGLE,
					  *data_offset_size, fragment, fragment_size);
```

Comment (jxstelter) [reply src/audio/module_adapter/module/iadk_modules.c:254]: fixed

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@jxstelter can you check CI. Thanks.

**Comment by lgirdwood** [comment]
@lyakh should we be seeing this ? (maybe a change in upstream Zephyr ?)
```
ccache /opt/sparse/bin/cgcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I/zep_workspace/zephyr/kernel/include -I/zep_workspace/zephyr/arch/xtensa/include -I/zep_workspace/zephyr/include -Izephyr/include/generated -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/common/include -I/zep_workspace/sof/zephyr/include -I/zep_workspace/sof/zephyr/../src/platform/intel/cavs/include -I/zep_workspace/sof/zephyr/../src/platform/tigerlake/include -I/zep_workspace/modules/hal/xtensa/include -I/zep_workspace/modules/hal/xtensa/zephyr/soc/intel_tgl_adsp -isystem /zep_workspace/zephyr/lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include-fixed -fno-strict-aliasing -O2 -imacros /zep_workspace/build-tgl/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always --sysroot=/home/sof_zephyr_docker_builder/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/xtensa-intel_s1000_zephyr-elf -D__CHECKER__ -Wno-attributes -imacros /zep_workspace/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fstrict-overflow -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/zep_workspace/sof/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/zep_workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/zep_workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -std=c99 -nostdinc -MD -MT zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -c /zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c
Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 305.
Use of uninitialized value $gccmachine in scalar chomp at /opt/sparse/bin/cgcc line 306.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 308.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 310.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 312.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 314.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 316.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 318.
Use of uninitialized value $gccmachine in string eq at /opt/sparse/bin/cgcc line 320.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 322.
Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 324.
Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 101.
Use of uninitialized value $gcc_base_dir in scalar chomp at /opt/sparse/bin/cgcc line 102.
Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 105.
Use of uninitialized value $multiarch_dir in scalar chomp at /opt/sparse/bin/cgcc line 106.
/zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c: note: in included file (through /zep_workspace/zephyr/include/zephyr/arch/xtensa/arch.h, /zep_workspace/zephyr/include/zephyr/arch/cpu.h, /zep_workspace/zephyr/include/zephyr/kernel_includes.h, ...):
/zep_workspace/modules/hal/xtensa/include/xtensa/config/core.h:54:10: error: unable to open 'core-isa.h'
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /zep_workspace/build-tgl
```

**Comment by jxstelter** [comment]
> @lyakh should we be seeing this ? (maybe a change in upstream Zephyr ?)
> 
> ```
> ccache /opt/sparse/bin/cgcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I/zep_workspace/zephyr/kernel/include -I/zep_workspace/zephyr/arch/xtensa/include -I/zep_workspace/zephyr/include -Izephyr/include/generated -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/common/include -I/zep_workspace/sof/zephyr/include -I/zep_workspace/sof/zephyr/../src/platform/intel/cavs/include -I/zep_workspace/sof/zephyr/../src/platform/tigerlake/include -I/zep_workspace/modules/hal/xtensa/include -I/zep_workspace/modules/hal/xtensa/zephyr/soc/intel_tgl_adsp -isystem /zep_workspace/zephyr/lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include-fixed -fno-strict-aliasing -O2 -imacros /zep_workspace/build-tgl/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always --sysroot=/home/sof_zephyr_docker_builder/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/xtensa-intel_s1000_zephyr-elf -D__CHECKER__ -Wno-attributes -imacros /zep_workspace/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fstrict-overflow -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/zep_workspace/sof/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/zep_workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/zep_workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -std=c99 -nostdinc -MD -MT zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -c /zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 305.
> Use of uninitialized value $gccmachine in scalar chomp at /opt/sparse/bin/cgcc line 306.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 308.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 310.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 312.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 314.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 316.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 318.
> Use of uninitialized value $gccmachine in string eq at /opt/sparse/bin/cgcc line 320.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 322.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 324.
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 101.
> Use of uninitialized value $gcc_base_dir in scalar chomp at /opt/sparse/bin/cgcc line 102.
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 105.
> Use of uninitialized value $multiarch_dir in scalar chomp at /opt/sparse/bin/cgcc line 106.
> /zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c: note: in included file (through /zep_workspace/zephyr/include/zephyr/arch/xtensa/arch.h, /zep_workspace/zephyr/include/zephyr/arch/cpu.h, /zep_workspace/zephyr/include/zephyr/kernel_includes.h, ...):
> /zep_workspace/modules/hal/xtensa/include/xtensa/config/core.h:54:10: error: unable to open 'core-isa.h'
> ninja: build stopped: subcommand failed.
> FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /zep_workspace/build-tgl
> ```

I think I see the same failure on all other open PR's. I dont think this is related with this particular code.

**Review by kv2019i — APPROVED** [review]
Looks good once we get CI results green.

**Comment by lyakh** [comment]
> @lyakh should we be seeing this ? (maybe a change in upstream Zephyr ?)
> 
> ```
> ccache /opt/sparse/bin/cgcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I/zep_workspace/zephyr/kernel/include -I/zep_workspace/zephyr/arch/xtensa/include -I/zep_workspace/zephyr/include -Izephyr/include/generated -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp -I/zep_workspace/zephyr/soc/xtensa/intel_adsp/common/include -I/zep_workspace/sof/zephyr/include -I/zep_workspace/sof/zephyr/../src/platform/intel/cavs/include -I/zep_workspace/sof/zephyr/../src/platform/tigerlake/include -I/zep_workspace/modules/hal/xtensa/include -I/zep_workspace/modules/hal/xtensa/zephyr/soc/intel_tgl_adsp -isystem /zep_workspace/zephyr/lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/bin/../lib/gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/include-fixed -fno-strict-aliasing -O2 -imacros /zep_workspace/build-tgl/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always --sysroot=/home/sof_zephyr_docker_builder/zephyr-sdk-0.15.1/xtensa-intel_s1000_zephyr-elf/xtensa-intel_s1000_zephyr-elf -D__CHECKER__ -Wno-attributes -imacros /zep_workspace/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fstrict-overflow -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/zep_workspace/sof/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/zep_workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/zep_workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -std=c99 -nostdinc -MD -MT zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -c /zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 305.
> Use of uninitialized value $gccmachine in scalar chomp at /opt/sparse/bin/cgcc line 306.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 308.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 310.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 312.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 314.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 316.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 318.
> Use of uninitialized value $gccmachine in string eq at /opt/sparse/bin/cgcc line 320.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 322.
> Use of uninitialized value $gccmachine in pattern match (m//) at /opt/sparse/bin/cgcc line 324.
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 101.
> Use of uninitialized value $gcc_base_dir in scalar chomp at /opt/sparse/bin/cgcc line 102.
> Can't exec "/opt/toolchains/zephyr-sdk-0.15.0/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc": No such file or directory at /opt/sparse/bin/cgcc line 105.
> Use of uninitialized value $multiarch_dir in scalar chomp at /opt/sparse/bin/cgcc line 106.
> /zep_workspace/zephyr/arch/xtensa/core/offsets/offsets.c: note: in included file (through /zep_workspace/zephyr/include/zephyr/arch/xtensa/arch.h, /zep_workspace/zephyr/include/zephyr/arch/cpu.h, /zep_workspace/zephyr/include/zephyr/kernel_includes.h, ...):
> /zep_workspace/modules/hal/xtensa/include/xtensa/config/core.h:54:10: error: unable to open 'core-isa.h'
> ninja: build stopped: subcommand failed.
> FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /zep_workspace/build-tgl
> ```

@lgirdwood isn't it what #6370 is supposed to fix?

**Comment by pjdobrowolski** [comment]
SOFCI TEST

**Review by pjdobrowolski — APPROVED** [review]

**Comment by kv2019i** [comment]
Sparse tool config issue not related to this PR, so proceeding with merge.

**Pull request merged**